src/utils/messages.ts

```diff
@@ -1,92 +1,39 @@
+import { isSameMinute } from 'date-fns';
 
+import { messageExtension } from './messageExtension';
 import { LOCAL_MESSAGE_CUSTOM_TYPE, type SuggestedMessageContent } from '../const';
 
-const TIME_SPAN = 3 * 60 * 1000;
-/**
- * Function to group messages based on their creation time
- *
- * @param {EveryMessage[]} messages - Array of messages to group
- * @returns {EveryMessage[]} - Array of messages grouped by creation time
- */
-export function groupMessagesByShortSpanTime(messages: BaseMessage[]): BaseMessage[] {
-  // Create an object to group messages based on their creation time
-  const groupedMessagesByCreatedAt = messages.reduce(
-    (groups, message, idx) => {
-      // Get the key of the previous group
-      const prevKey = Object.keys(groups)[Object.keys(groups).length - 1];
-
-      // Check if the time difference between the current message and the previous one is within 3 minutes
-      if (
-        prevKey &&
-        message.createdAt - Number(prevKey) <= TIME_SPAN &&
-        !message.isAdminMessage() &&
-        getSenderUserIdFromMessage(message) === getSenderUserIdFromMessage(messages[idx - 1])
-      ) {
-        // Add the message to the existing group
-        return {
-          ...groups,
-          [prevKey]: [...(groups[prevKey] ?? []), message],
-        };
-      } else {
-        // Create a new group for the current message
-        return {
-          ...groups,
-          [message.createdAt]: [message],
-        };
-      }
-    },
-    {} as Record<string, BaseMessage[]>,
-  );
-
-  // Flatten the grouped messages and add chain indicators
-  return Object.values(groupedMessagesByCreatedAt).flatMap((messages: BaseMessage[]) => {
-    if (messages.length > 1) {
-      // Add chain indicators to the first and last messages in the group
-      messages.forEach((message, index) => {
-        // FIXME: Remove data pollution.
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore
-        message.chainTop = index === 0;
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore
-        message.chainBottom = index === messages.length - 1;
-      });
-
-      return messages;
-    }
-    return messages;
-  });
-}
-
-export function getBotWelcomeMessages(messages: BaseMessage[], botUserId: string | null) {
-  // if the list is empty or the first message is not from bot,
-  // we just assume there's no welcome messages
-  if (messages.length === 0 || getSenderUserIdFromMessage(messages[0]) !== botUserId) {
-    return [];
+export const getMessageGrouping = (curr: BaseMessage, prev?: BaseMessage, next?: BaseMessage): [boolean, boolean] => {
+  if (!curr.isUserMessage() && !curr.isFileMessage()) {
+    return [false, false];
   }
 
-  // if the list has only bot messages, then just return the whole list
-  if (messages.every((message) => getSenderUserIdFromMessage(message) === botUserId)) {
-    return messages;
-  }
+  const getTop = () => {
```

Comment:
can we change the name of the function? it suggests returning top but the function returns boolean.